### PR TITLE
Enable suppressing all output ("silent mode")

### PR DESCRIPTION
**Pull request by markhaehnel**
This solves issue #590.

I'm not exactly sure if it's the right approach but it keeps all original functionality without breaking changes and just adds a option to enable silent mode.
Silent mode just ignores all logging calls.

I tried to keep it similar to the `verbose` and `extra_verbose` functionality.

Silent mode is not exposed in the CLI options because i don't really see a need for it right now.

```rust
// Example on how to run it in silent-mode now

let current = Rustdoc::from_root(path_to_project_root);
let mut check = Check::new(current);
check.with_silent(true);

let result = check.check_release();
```

### CODE REVIEW

**Comment by markhaehnel** [comment]
> The second option would be to repurpose the log_level: None niche in the GlobalConfig struct to mean "no logging at all" instead of its current meaning of "no explicit value was requested, use the default." This would require some more careful refactoring to ensure that we don't accidentally drop log output in the current configuration.

I've gone this route. Was relatively simple while keeping all logging statements as is.
Please take a look and give feedback. :)

> If you're interested in more improvements around this, consider in a separate PR removing the need for config.stdout() and config.stderr() methods by making a colored_ln() method on config which can internally check that logging at the correct level is enabled. Right now, the config.log(|config| colored_ln(config.stdout(), ...)) blocks are getting a bit too involved to be maintainable, and I worry they'll be bug-prone.

Regarding this one i will open an new PR in the next days when i find time.

**Review by obi1kenobi — COMMENTED** [review]
The implementation broadly looks good.

There are possibly a few bugs and risk points where it'd be good to double-check the docs / add test cases / add docs on our newly-changed public API. Nothing major, probably all resolvable in the next iteration on this PR.

File: src/config.rs [hunk 1]
```diff
@@ -69,6 +69,10 @@ impl GlobalConfig {
         self
     }
 
+    pub fn is_info(&self) -> bool {
+        log::Level::Info <= self.level.unwrap_or(log::Level::Error)
+    }
+
     pub fn is_verbose(&self) -> bool {
         log::Level::Debug <= self.level.unwrap_or(log::Level::Error)
     }
```
[the remark this answers is not in the document]
Comment (markhaehnel) [reply src/config.rs]: I've got those log levels under control now and added some tests! Behaviour is now exactly as it was when using (-v, -vv, -vvv, -q etc.)

File: src/main.rs [hunk 2]
```diff
@@ -319,9 +319,9 @@ impl From<CheckRelease> for cargo_semver_checks::Check {
         if let Some(baseline) = custom_baseline {
             check.with_baseline(baseline);
         }
-        if let Some(log_level) = value.verbosity.log_level() {
-            check.with_log_level(log_level);
-        }
```
[the remark this answers is not in the document]
Comment (markhaehnel) [reply src/main.rs:324]: The log level here is only for the CLI options. Default is Info as set in the `CheckRelease` struct.

```rust
// src/main.rs(260, 0)
    #[command(flatten)]
    verbosity: clap_verbosity_flag::Verbosity<clap_verbosity_flag::InfoLevel>,
```

**Comment by markhaehnel** [comment]
Thanks for guiding me through this and for spotting the issue with the log levels that i missed.
Feel free to give some more feedback if something is not as you would have liked it! 😊

File: src/lib.rs [hunk 3]
```diff
@@ -37,6 +37,7 @@ pub struct Check {
     release_type: Option<ReleaseType>,
     current_feature_config: rustdoc_gen::FeatureConfig,
     baseline_feature_config: rustdoc_gen::FeatureConfig,
+    silent: bool,
```
Comment (obi1kenobi) [inline src/lib.rs:40]: This is presumably just a leftover and should be removed, right?
```suggestion
```

File: src/lib.rs [hunk 4]
```diff
@@ -235,6 +236,7 @@ impl Check {
             release_type: None,
             current_feature_config: rustdoc_gen::FeatureConfig::default_for_current(),
             baseline_feature_config: rustdoc_gen::FeatureConfig::default_for_baseline(),
+            silent: false,
```
Comment (obi1kenobi) [inline src/lib.rs:239]:
```suggestion
```

**Comment by obi1kenobi** [comment]
> I enjoyed helping out! Thank you too!

I'm glad to hear that 😁 If you're open to contributing more, I'd love that and I'd love to help! I can even suggest things that are easy to dive into without getting too tangled up.

**Comment by obi1kenobi** [comment]
Try this one: https://github.com/obi1kenobi/cargo-semver-checks/issues/311

I'd split it into a few parts. First, add an `EnvironmentVariables` [collector](https://crates.io/crates/bugreport) for [the target dir](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir) and for [rustflags](https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags) environment variables. This should be a super easy win and will give us info to triage some common bugs.

There are more ways to specify cargo configuration. So as a stretch goal, see if you can figure out a way the configuration cargo is _actually_ using, even if it wasn't set via an env var. Reading some of the cargo docs would be helpful here, I'm not exactly sure what the options we have are — I'd be relying on you to figure out whether there's a good path forward here (and if so, what it is), or if it's not worth it. Once you've made up your mind, let's talk about it before you write any code for it — it will make the code review easier.

For bonus points, you could also try to generate a URL that will automatically open and pre-populate a GitHub issue with all the relevant info for the bug the user is trying to report: https://github.com/obi1kenobi/cargo-semver-checks/issues/602

**Comment by markhaehnel** [comment]
> Also, I just released v0.26 with this PR 🎉

Great!




> Try this one: #311
> 
> I'd split it into a few parts. First, add an `EnvironmentVariables` [collector](https://crates.io/crates/bugreport) for [the target dir](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir) and for [rustflags](https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags) environment variables. This should be a super easy win and will give us info to triage some common bugs.
> 
> There are more ways to specify cargo configuration. So as a stretch goal, see if you can figure out a way the configuration cargo is _actually_ using, even if it wasn't set via an env var. Reading some of the cargo docs would be helpful here, I'm not exactly sure what the options we have are — I'd be relying on you to figure out whether there's a good path forward here (and if so, what it is), or if it's not worth it. Once you've made up your mind, let's talk about it before you write any code for it — it will make the code review easier.
> 
> For bonus points, you could also try to generate a URL that will automatically open and pre-populate a GitHub issue with all the relevant info for the bug the user is trying to report: #602

I will take a look at this when i have some free time next week!

**Comment by obi1kenobi** [comment]
Awesome! I'm on the Rust Zulip if you need anything that isn't a good fit for GitHub issues: https://rust-lang.zulipchat.com/

**Pull request merged**